WIP/BLD: daff perms

tangkong opened the pull request:
- Transition comment diff workflow to trigger on `pull_request` to lock down write permission tokens
- Try using github-scripts for posting comments

Maybe:
scope out a token with comment permissions and use that for the script specifically

tangkong commented:
sigh, secrets are not passed when pull requests are made from forked repositories.  `pull_request_target` may be the what we have to go with here. 

I wanted to use github-scripts for the sake of using an action that's supported by github explicitly, but it should be functionally identical to the tholliander workflow

tangkong commented:
Interesting, I guess I should have tried that first.  😆

tangkong commented:
This is again something that will require a merge to test

tangkong commented:
In some more concise terms, we're splitting this up into two workflows in order to give each part of the process the most restrictive set of permissions:
* checking out the code and generating the diff: READ-ONLY
* Writing the comment: Write-PR permissions only

By splitting this up we then require passing the diff information through artifacts, but it allows us to more strictly limit permissions.

`pull_request_target` gives **read and write** permissions, and should not be used in workflows that checkout PR contents (which can contain malicious code).  This is balanced by `pull_request_target` triggered workflows only running based on the base of the PR (not the modified code), but it is possible to circumvent this (I did this at an earlier part of this commit history)

Mostly putting this into words for my own education, but maybe also for the education of others.

ZLLentz commented on .github/workflows/makecomment.yml:
```diff
@@ -0,0 +1,49 @@
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: issue_number,
+              body: 'Everything is OK. Thank you for the PR!'
```
I'm confused, where is the part where we write out the diff to a comment?
I'm on board with learning and applying best-practices for security.

tangkong replied:
I need to stop working on this in the morning.  💀

tangkong commented on .github/workflows/daff.yml:
```diff
@@ -32,15 +32,23 @@ jobs:
         pip install daff
 
     - name: make diff
+      id: make_diff
```
This was from when I was trying to store the diff results as a step output, I needed a step ID so I could reference that output specifically.  This should also go now that we split up the workflows

tangkong commented:
Thanks for the careful eye, I've clearly not been careful enough qa'ing this on my own.  

Reading over this again I think might want some escape hatch for when there is no diff generated, in the case where commits are added that don't affect the csvs.  Maybe that doesn't matter 🤷 

I think I addressed all the comments 🤞

tangkong commented:
Ah no you're right, that should hopefully be sufficient

ZLLentz reviewed (approved):
This looks good to me. Maybe it's time to merge and check the result in a follow-up.

tangkong commented:
Merging, thanks for all the QA and sorry for the sloppiness throughout (I'll do better).  

Turns out the required checks needed exactly the name of the job as defined in the yaml, not the name as displayed in the check itself.  ("generate-xml", not "Process on PR / generate-xml").  This conflicts with my past experience with pcds-ci-helpers, but I won't lose any sleep
